### PR TITLE
Whitelist Length Control

### PR DESCRIPTION
**Pull request by 0xfps**
## Whitelist Length Control.

Added the **Whitelist Length Control Interface**, required according to issue #7 .

## Environment

- Compiler version: 0.8.7
- Target EVM version (as per compiler settings): default
- Framework/IDE (e.g. Truffle or Remix): Remix

- Operating system: Windows

## Whitelist length control.

For contracts that will implement this interface, storage fixed arrays cannot be created from a function (only memory can, and will be wiped out when the function is done). But this interface seeks to control a dynamic array's max length using a storage uint256 variable. This ensures that the length of the array cannot be > the value of the max length. Contracts to implement this interface must specify a control uint256 value on the contract storage.

###
Anthony.

### CODE REVIEW

File: src/contracts/interfaces/IWhitelist.sol [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+* This max length value can only be incremented.
+*/
+
+interface IWhitelist
+{
+    // ========== E V E N T S ==========
+
+    // Emitted when a new whitelist length is set.
+    // (uint256) => Value of the max length.
+    event SetLength(uint256);
```
Comment (koolamusic) [inline src/contracts/interfaces/IWhitelist.sol]: make the events move descriptive so we can do something like `SetWhitelistLength` and `ExtendWhitelistLength`

File: src/contracts/interfaces/IWhitelist.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    * Returns the current value of the max length of the array.
+    */
+    function getMaxLength() external returns(uint256);
+
+
+    /*
+    * @dev:
+    * Set the new `_length` to  the max length of the array.
+    * This cannot be reduced, only increased.
+    * (An array with a current max length of 5, can only accept values >= 6 for a new max length value).
```
Comment (koolamusic) [inline src/contracts/interfaces/IWhitelist.sol]: Let's also stripe to keep implementations for LTE and GTE to just use `GT` or `LT` 
Greater than or less than

**Review by koolamusic — CHANGES_REQUESTED** [review]

**Comment by koolamusic** [comment]
Also add something in the comments about usability like.
In order to use this in memory we can do something like `uint8[] memory whitelist = new uint8[](MAX_WLL)`

an idea 💡

**Comment by koolamusic** [comment]
Also, so that you get the full drift. 
The whole idea behind this is that we can have `free` accounts on Daccred.

These free accounts can be limited to 100 users max or 1000. Where they will now need to pay a fee to extend their WL length. 

So in context when we use this extension, we can actually deploy a contract with constructor arguments to fix the length of their whitelist. 

A user who pays has unlimited access & unlimited freedom.

File: src/contracts/interfaces/IWhitelist.sol (hunk 1 above)
Comment (0xfps) [reply src/contracts/interfaces/IWhitelist.sol]: I will do just that 👍🏾.

File: src/contracts/interfaces/IWhitelist.sol (hunk 2 above)
Comment (0xfps) [reply src/contracts/interfaces/IWhitelist.sol]: Alright, then.

**Comment by 0xfps** [comment]
> 

I will work on this.

**Comment by 0xfps** [comment]
I have added the extra usability function on how the Interface works when implemented with the future WhitelistFactory.

**Comment by koolamusic** [comment]
LGTM

**Pull request merged**